core/src/main/scala/cilib/Archive.scala

```diff
@@ -0,0 +1,195 @@
+package cilib
+
+import scalaz._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.numeric.Positive
+
+sealed trait Bound
```

Comment:
This structure should potentially be hidden - it's specific to this archive implementation - if an unbounded archive should be created, then a better way would be to have a constructor function:

`Archive.unbounded()` and `Archive.bounded(n)`

Or something similar.